chronos/asyncloop.nim

```diff
@@ -1027,11 +1019,15 @@ proc wait*[T](fut: Future[T], timeout = InfiniteDuration): Future[T] =
             when T is void:
               retFuture.complete()
             else:
-              retFuture.complete(fut.read())
+              try:
+                retFuture.complete(fut.read())
+              except CatchableError as exc:
+                retFuture.fail(exc)
```

Comment:
This should be `Defect`. Because this situation is not possible. There present check at https://github.com/status-im/nim-chronos/pull/166/files#diff-95090cf0d6917b2c1eae2c05fe84b275d0f77b28ac5f0638651bd2d672e756f8R1016 which protects reading `fut` value from raising exception.